src/prefect/utilities/notifications/notifications.py

```diff
@@ -318,3 +318,136 @@ def add(x, y):
     if not r.ok:
         raise ValueError("Slack notification for {} failed".format(tracked_obj))
     return new_state
+
+
+def snowflake_message_formatter(
+    tracked_obj: TrackedObjectType,
+    state: "prefect.engine.state.State",
+) -> dict:
+    fields = []
+    if isinstance(state.result, Exception):
+        value = "{}".format(repr(state.result))
+    else:
+        value = cast(str, state.message)
+    if value is not None:
+        fields.append({"value": value.replace("'", "''")})
+
+    notification_payload = {
+        "flow_id": prefect.context.get("flow_run_id"),
+        "flow_name": prefect.context.get("flow_name"),
+        "task_name": tracked_obj.name,
+        "state": type(state).__name__,
+        "message": fields,
+    }
+
+    return notification_payload
+
+
+@curry
+def snowflake_logger(
+    tracked_obj: TrackedObjectType,
+    old_state: "prefect.engine.state.State",
+    new_state: "prefect.engine.state.State",
+    ignore_states: list = None,
+    only_states: list = None,
+    snowflake_secret: str = None,
+    snowflake_log_table_name: str = None,
+    test_env: bool = False,
+) -> "prefect.engine.state.State":
+    """
+    Snowflake state change handler/logger; requires having the Prefect Snowflake app installed.
+    Works as a standalone state handler, or can be called from within a custom state handler.  This
+    function is curried meaning that it can be called multiple times to partially bind any
+    keyword arguments (see example below).
+    Args:
+        - tracked_obj (Task or Flow): Task or Flow object the handler is
+            registered with
+        - old_state (State): previous state of tracked object
```

Comment:
```suggestion
        - old_state (State): Previous state of tracked object.
```

Reply:
Docstrings should be sentences, starting capital and closing period. Examples above.